src/pty.rs

```diff
@@ -0,0 +1,72 @@
+use std::ffi::CStr;
+use std::os::unix::io::RawFd;
+
+use libc;
+
+use {Errno, Error, fcntl, Result};
+
+pub fn grantpt(fd: RawFd) -> Result<()> {
+    if unsafe { libc::grantpt(fd) } < 0 {
+        return Err(Error::last().into());
+    }
+
+    Ok(())
+}
+
+pub fn posix_openpt(flags: fcntl::OFlag) -> Result<RawFd> {
+    let fd = unsafe {
+        libc::posix_openpt(flags.bits())
+    };
+
+    if fd < 0 {
+        return Err(Error::last().into());
+    }
+
+    Ok(fd)
+}
+
+pub fn ptsname(fd: RawFd) -> Result<String> {
+    let name_ptr = unsafe { libc::ptsname(fd) };
+    if name_ptr.is_null() {
+        return Err(Errno::ENOTTY.into());
```

Comment:
Is Err(Error::Sys(Errno::last())) not safer here? I mean according to standard ENOTTY is the only error, but you can never be sure.